ingestion/src/metadata/ingestion/models/table_metadata.py

```diff
@@ -158,6 +158,37 @@ class PipelineESDocument(BaseModel):
     doc_as_upsert: bool = True
 
 
+class UserESDocument(BaseModel):
+    """Elastic Search Mapping doc for Users"""
+
+    user_id: str
+    deleted: bool
+    entity_type: str = "user"
+    name: str
+    display_name: str
+    email: str
+    suggest: List[dict]
+    last_updated_timestamp: Optional[int]
+    teams: List[str]
+    roles: List[str]
+    doc_as_upsert: bool = True
+
+
+class TeamESDocument(BaseModel):
```

Comment:
Would it make sense to prepare these models as part of the JSON schemas as well? I see that in the resources we are defining some mapping files, whose properties do not fully match this class def here.

I would assume that if we now need to make a change in the mapping, we'd need to update this class as well as the constant below (e.g., `USER_ELASTICSEARCH_INDEX_MAPPING`).

Centralising this info would make the management easier and we could directly have:

```python
USER_ELASTICSEARCH_INDEX_MAPPING = {
    "mappings": MyClass.schema()
}
```

Reply:
@pmbrull Agree. This is in back off my mind and didn't get to it. Filed a ticket for next release https://github.com/open-metadata/OpenMetadata/issues/2753